services/api/src/resources/deployment/resolvers.ts

```diff
@@ -193,21 +193,20 @@ export const getDeploymentsByFilter: ResolverFn = async (
   }
 
   let queryBuilder = knex.select("deployment.*").from('deployment').
-      join('environment', 'deployment.environment', '=', 'environment.id').
-      join('project', 'environment.project', '=', 'project.id').
-      where('environment.deleted', '=', '0000-00-00 00:00:00');
+      join('environment', 'deployment.environment', '=', 'environment.id');
+
+  if (userProjectIds) {
+      queryBuilder.join('project', 'environment.project', '=', 'project.id').
```

Comment:
Actually, do we even pull anything from `project`?

We might want to kill this join directly, don't you think? Then we just say (below) 
`queryBuilder = queryBuilder.whereIn('environment.project', userProjectIds);`

Reply:
You're right, we don't need to join at all. Will update